### ISSUE TITLE
[bug]Breadcrumbed only work for the first time I open my computer and the first time i open obsidian 

### ISSUE THREAD

**Issue by Ravenclawer** [SkepticMystic/breadcrumbs#83]
I am on windows 1903 and my obsidian version was 0.12.12, after creating sets of notes according to breadcrumbed plugin wiki, the breadcrumbed matrix didn't show what it was supposed to display.
![image](https://user-images.githubusercontent.com/73892411/128956176-d4fb009e-1595-4966-8c4a-1351f5e3246d.png)
and the visualizer didn't display anything but an empty frame.
![image](https://user-images.githubusercontent.com/73892411/128956246-2f64e322-e592-4f22-8c17-e3344f79a275.png)
  these issues could not be solved by restart obsidian but could be temporarily solved by reboot the system and restart obsidian then if you revise anything, the breadcrumbed plugin will be acting up. 
I have opened the debugging mode and nothing happened 
![image](https://user-images.githubusercontent.com/73892411/128956598-fe12fb6e-30f0-4404-9723-824bf186f608.png)

**Comment by Ravenclawer** [comment]
the breadcrumbed plugin was 0.10.11

**Comment by Ravenclawer** [comment]
I have seen a similar issue in the closed section and that issue seems to be solved by upgrading obsidian to 0.10.5, maybe this issue will solve after the next upgrade of obsidian software.

**Comment by SkepticMystic** [comment]
Hey RavenClawer, thanks for posting the bug.
Can you keep super debug mode enabled, open up the Obsidian console (Ctrl + Shift + I), change to a different note, and then send the last few console logs that appear.

Also, do you get any errors in the console before doing super debug?

**Comment by Ravenclawer** [comment]
this is the first few lines in the console after i opened Obsidian 
![image](https://user-images.githubusercontent.com/73892411/128980335-77192d55-0e78-4cc0-b5e2-60e826dca261.png)
![image](https://user-images.githubusercontent.com/73892411/128980379-febd35f5-a41b-43d5-b284-510dc97bb66c.png)


this is the last few lines in the console
![image](https://user-images.githubusercontent.com/73892411/128980191-37bc3598-309c-462a-b2a9-9579b4a1f683.png)

**Comment by SkepticMystic** [comment]
Hmmm ok, that's strange. Can you turn off super debug, enable regular debug mode, change to a different note, and then look inside the `fileFrontmatterArr` object to see if things look right. You should see any `undefined`

**Comment by Ravenclawer** [comment]
I turn off the supper debug mode and it appears as follow:
![image](https://user-images.githubusercontent.com/73892411/128984526-b7ded791-6379-4083-8b1f-df6ea89e9f81.png)

**Comment by Ravenclawer** [comment]
I turn off the supper debug mode and it appears as follow:
![image](https://user-images.githubusercontent.com/73892411/128984526-b7ded791-6379-4083-8b1f-df6ea89e9f81.png)

**Comment by Ravenclawer** [comment]
it seems like it had picked up the metadata:
![image](https://user-images.githubusercontent.com/73892411/128997065-6a3a390a-bbf1-4db3-87c8-f72974601e05.png)

**Comment by SkepticMystic** [comment]
Are any of the entries in `fileFrontmatterArr` `undefined`?

**Comment by Ravenclawer** [comment]
I do have many files other than markdown format in the vault, maybe those files caused the problem. as to entries, they all got some values, but some of them have some information even when you collapse them.
![image](https://user-images.githubusercontent.com/73892411/129001767-cfa089ce-2fb5-4654-b602-914ec89d44a5.png)
and some of them includes strange information.
![image](https://user-images.githubusercontent.com/73892411/129001691-5991925a-50d5-4c36-b53d-32ded598ee40.png)

**Comment by SkepticMystic** [comment]
@Ravenclawer can you give it a try on the latest version and let me know if anything has changed

**Comment by Ravenclawer** [comment]
I have upgraded the Breadcrumbed plugin to 0.10.21, and it still won't work.
Thank you so much for working on this, I have noticed there is a refresh button you have added. I believe it was caused by the large number of plugins that  I have installed and some unusual file types that I put in the vault(i have put the entire Zotero program in the vault to make linking to pdfs in Zotero convenient but also bring in lots of files other than markdown).
Here is the snapshot my current note  page with breadcrumbed enabled:
![image](https://user-images.githubusercontent.com/73892411/129372499-7bb16953-e674-4832-bcc8-f6ff546e0466.png)
Here is the debug page
![image](https://user-images.githubusercontent.com/73892411/129372949-420eac5f-b57d-4902-a762-e1097a253072.png)
for the previous version, I could have it worked the first time I reboot windows and then open the obsidian software. but now, after reboot, the breadcrumbed plugin doesn't work. I don't know if it's the cMenu plugin that I have incooperate into the vault that caused this.

**Comment by SkepticMystic** [comment]
I think this would be alot easier to debug if we had a Zoom call or something. Maybe over Discord?
Let me know if you're open to that, my Discord handle is @SkepticMystic

**Comment by Ravenclawer** [comment]
Thank you, you're so kind, today. I reinstalled obsidian and try to create new vaults to try on different versions of breadcrumbed. I found that this issue is resolved in the 0.10.22 version of the plugin. 
The following is my problem-solving process.
I am using the Tokyo night theme  for my vault, and the setting page of breadcrumbed appeared like this:
![image](https://user-images.githubusercontent.com/73892411/129440219-b9267715-c912-4ffc-b3ee-5612fc403c66.png)
So at the first, I didn't see little arrows in the slots for defining parent, sibling, and child. 
when I open the setting in a new vault, I can see the inside more clearly that it contains an arrow in each slot
![image](https://user-images.githubusercontent.com/73892411/129440313-1cb2f592-d17e-4fbb-9c86-1504c52703b0.png)

and after when through some trail and errors with different versions of breadcrumbed plugins and different ways in putting the field in each slot like:  <↑parent>, <↑parent::>, <↑upper>, <↑upper::>, etc. I found there is no need to keep ↑ when you define the field for the hierarchy, and just delete it and put word parent in that slot could do the magic.
now the plugin works pretty well in my old and new vaults:
Tokyo night theme 
![image](https://user-images.githubusercontent.com/73892411/129440546-268c8be2-78b4-4e3a-bf49-dcb04cda6d6b.png)
default theme 
![image](https://user-images.githubusercontent.com/73892411/129440517-96fe8566-c219-421f-9022-331ba2147de7.png)
